### ISSUE TITLE
locate does not work under Chrome when a watchPosition in on

### ISSUE THREAD

**Issue by rbaprado** [domoritz/leaflet-locatecontrol#108]
It seems that when a watchPosition is active, the locate method hangs forever under Chrome (switching tabs seem to solve the problem). I tested under Firefox, Chrome for Android and IE 11, the issue is only happening on desktop Chrome.

This guy seems to have find the same problem (not on leaflet though): http://stackoverflow.com/questions/21096846/chrome-getcurrentposition-timeout-after-watchposition-was-called/27071329#27071329

**Comment by jailby** [comment]
I just installed leaflet-locatecontrol for one of the project I'm working on and I have the same issue.

Anyone else has the same problem? 

This issue was posted a year ago, so I'm wondering if there is any plan on fixing this issue?

**Comment by barbeau** [comment]
cc @jmfield2

**Comment by domoritz** [comment]
I'm looking into it now.

**Comment by domoritz** [comment]
I'm having troubles to replicate this issue. How is a watchPosition active? I am using watch position already in https://github.com/domoritz/leaflet-locatecontrol/blob/gh-pages/src/L.Control.Locate.js#L91. 

I've tested on desktop Chrome 46.0.2490.33 beta (64-bit) on Mac OS with the demo from this repo. If you have this issue, please report your versions and the exact code that I can use to replicate this problem.

**Comment by jailby** [comment]
The demo works fine for me.

I have added this project to the following:
https://github.com/CUTR-at-USF/usf-mobullity/

I added the project through the mapbox CDN [here](https://github.com/CUTR-at-USF/usf-mobullity/blob/mobullityrebase/src/client/index.html#L38).

Then when I tested with Chrome and Firefox I noticed the bug described by @rbaprado 
Google Chrome   43.0.2357.132 (Official Build) (64-bit)

Firefox works but Chrome stall and changing tab solve the problem.

**Comment by barbeau** [comment]
@jmfield2 and @jailby can you please deploy this control to http://mobullity.forest.usf.edu/ so @domoritz can see the problem there?

And please share the exact code deployed that includes this control also.

**Comment by jailby** [comment]
@domoritz you can go and check the following project: http://mobullity.forest.usf.edu/

I tested it on multiple Chrome version including the one mentioned above and I replicate the issue each times.

[Here](https://github.com/CUTR-at-USF/usf-mobullity/blob/mobullityrebase/src/client/index.html#L40) you can see the inclusion via mapbox CDN and [here](https://github.com/CUTR-at-USF/usf-mobullity/blob/mobullityrebase/src/client/js/otp/core/Map.js#L168) the declaration of the button.

If you need more details to understand the problem let me know.

**Comment by jmfield2** [comment]
If it helps, we seem to have determined the cause to be a strange interaction between desktop chrome and watchPosition while another watchPosition is currently active - that is, the second watchPosition will stall until the user switches tabs.  

If you attempt to L.locate({watch:true}) and then click the leaflet location control (which also calls L.locate({watch:true}), you should see the stall on chrome.  This is what is happening on the mobullity site.

The fix would seem to be for leaflet to track the watch ID returned by navigator.watchPosition when watch=true and to clearWatch upon subsequent calls or by calling the already defined 'stopLocate' here:
https://github.com/Leaflet/Leaflet/blob/8c8d0ac7ee40d73036e119045b284b248dec3e39/src/map/ext/Map.Geolocation.js#L31

So, it appears like this may actually be a bug in leaflet and not with your control.. hopefully it helps someone! :)

**Comment by jmfield2** [comment]
Note, there is an open chromium issue for this @ 
https://code.google.com/p/chromium/issues/detail?id=452549&q=watchPosition&colspec=ID%20Pri%20M%20Stars%20ReleaseBlock%20Cr%20Status%20Owner%20Summary%20OS%20Modified

**Comment by domoritz** [comment]
Okay, can you send me a smaller example that shows the problem. If I understand correctly and you are creating two watchLocation, please try not to do that. Just add one location watcher and subscribe to the locate event on the map. 

This problem doesn't seem to be a problem with my code so I can try to help you resolve it but I won't need to change any code. I mark this as a question.

**Comment by jmfield2** [comment]
Hmm well we are using a watch position from leaflet in one area and then trying to deploy your control which will attempt a second watch position if the user clicks it. Is there an easy way to have a locate button on the map as in your Ctrl but feed it the location if we are already receiving pos updates?

A smaller example I think is the jsfiddle http://jsfiddle.net/Js6sm/  that might help you understand the core issue better

**Comment by domoritz** [comment]
> Is there an easy way to have a locate button on the map as in your Ctrl but feed it the location if we are already receiving pos updates?

No. But you can simply only use my control. If you want to hide it, use a custom layer that is not attached to the map. This way you only have one watch location.

**Comment by KingReload** [comment]
I solved the problem:

1. download the libraries `L.Control.Locate.min.css` & `L.Control.Locate.min.js`.
2. I imported the `https://unpkg.com/leaflet@1.0.3/dist/leaflet.js` library and linked the downloaded libraries to the html / php, so they are usable.
3. switch the browser connection to `https` and don't use `http`, because http is unsecured and chrome won't accept tracking through an unsecured address.
4. I used the `var lc = L.control.locate().addTo(map);` to locate my position on the map and it works totally fine in all browsers now.

Hope this solves the problem for more people that try to visit this issue in the future.
